catalog/core/catalog-core-validationparser/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -296,6 +367,42 @@ private void parseValidators(Map<String, Object> root, Outer outer) {
                 .collect(toSet());
     }
 
+    @SuppressWarnings("unchecked")
+    private MetacardValidator getMetacardValidator(
+            MetacardValidatorDefinition validatorDefinition) {
+        switch (validatorDefinition.validatorName) {
+        case REQUIRED_ATTRIBUTE_VALIDATOR_PROPERTY: {
+            String metacardType = null;
+            Object metacardTypeValue = validatorDefinition.arguments.get(METACARD_TYPE_PROPERTY);
+
+            if (metacardTypeValue instanceof String) {
+                metacardType = (String) metacardTypeValue;
+            }
+
+            List<String> requiredAttributes = new ArrayList<>();
+            Object requiredAttributesObj = validatorDefinition.arguments.get(
+                    REQUIRED_ATTRIBUTES_PROPERTY);
+            if (requiredAttributesObj instanceof List) {
+                List<Object> requiredAttrObjList = (List) requiredAttributesObj;
+                requiredAttributes = requiredAttrObjList.stream()
+                        .filter(String.class::isInstance)
+                        .map(String.class::cast)
+                        .collect(Collectors.toList());
+            }
+
+            if (metacardType != null && CollectionUtils.isNotEmpty(requiredAttributes)) {
+                Set<String> requiredAttrSet = new HashSet<>(requiredAttributes);
+                return new RequiredAttributesMetacardValidator(metacardType, requiredAttrSet);
+            } else {
```

Comment:
nit: No need for an else since the control flow is redirected (by the return) in the if case. In fact you could just use the fallthrough to throw the exception if you wish. (if you wish to keep the exception since its more specific thats cool too 👍 )